cmd/gateway-azure.go

```diff
@@ -592,66 +620,64 @@ func (a *azureObjects) PutObjectPart(bucket, object, uploadID string, partID int
 
 // ListObjectParts - Use Azure equivalent GetBlockList.
 func (a *azureObjects) ListObjectParts(bucket, object, uploadID string, partNumberMarker int, maxParts int) (result ListPartsInfo, err error) {
-	result.Bucket = bucket
-	result.Object = object
-	result.UploadID = uploadID
-	result.MaxParts = maxParts
-
-	if meta := a.metaInfo.get(uploadID); meta == nil {
-		return result, nil
-	}
-	resp, err := a.client.GetBlockList(bucket, object, storage.BlockListTypeUncommitted)
-	if err != nil {
-		return result, azureToObjectError(traceError(err), bucket, object)
-	}
-	tmpMaxParts := 0
-	partCount := 0 // Used for figuring out IsTruncated.
-	nextPartNumberMarker := 0
-	for _, part := range resp.UncommittedBlocks {
-		if tmpMaxParts == maxParts {
-			// Also takes care of the case if maxParts = 0
-			break
-		}
-		partCount++
-		partID, _, md5Hex, err := azureParseBlockID(part.Name)
-		if err != nil {
-			return result, err
-		}
-		if partID <= partNumberMarker {
-			continue
-		}
-		result.Parts = append(result.Parts, PartInfo{
-			partID,
-			UTCNow(),
-			md5Hex,
-			part.Size,
-		})
-		tmpMaxParts++
-		nextPartNumberMarker = partID
-	}
-	if partCount < len(resp.UncommittedBlocks) {
-		result.IsTruncated = true
-		result.NextPartNumberMarker = nextPartNumberMarker
+	metadataObject := fmt.Sprintf(metadataObjectTemplate, uploadID)
+	if _, err = a.client.GetBlobMetadata(bucket, metadataObject); err != nil {
+		errorIf(err, "unable to read metadata object for upload ID %s", uploadID)
+		return result, traceError(InvalidUploadID{})
 	}
 
+	// It's decided not to support List Object Parts, hence returning empty result.
 	return result, nil
 }
 
 // AbortMultipartUpload - Not Implemented.
 // There is no corresponding API in azure to abort an incomplete upload. The uncommmitted blocks
 // gets deleted after one week.
-func (a *azureObjects) AbortMultipartUpload(bucket, object, uploadID string) error {
-	a.metaInfo.del(uploadID)
-	return nil
+func (a *azureObjects) AbortMultipartUpload(bucket, object, uploadID string) (err error) {
+	metadataObject := fmt.Sprintf(metadataObjectTemplate, uploadID)
+	if _, err = a.client.GetBlobMetadata(bucket, metadataObject); err != nil {
+		errorIf(err, "unable to read metadata object for upload ID %s", uploadID)
+		return traceError(InvalidUploadID{})
+	}
+
+	err = a.client.DeleteBlob(bucket, metadataObject, nil)
+
+	return err
```

Comment:
can be one line:
`return a.client.DeleteBlob(bucket, metadataObject, nil)`

Reply:
Done